pkg/plugin/plugin.go

```diff
@@ -633,8 +653,21 @@ func CmdCheck(args *skel.CmdArgs) error {
 	}
 	ovsHWOffloadEnable := sriov.IsOvsHardwareOffloadEnabled(netconf.DeviceID)
 
+	// check cache
+	cRef := config.GetCRef(args.ContainerID, args.IfName)
+	cache, err := config.LoadConfFromCache(cRef)
+	if err != nil {
+		return err
+	}
+
+	if err := validateCache(cache, netconf); err != nil {
```

Comment:
Done. It was my mistake. Thanks for your comment.
I moved the IPAM check below the cache validation, instead of moving the cache validation up.

https://github.com/k8snetworkplumbingwg/ovs-cni/blob/05df2fe90ffb56acd80aac322e207689481b9fb7/pkg/plugin/plugin.go#L690-L698